kpm/lib/kpm/system.rb

```diff
@@ -0,0 +1,166 @@
+
+module KPM
+
+  module OS
+    def OS.windows?
+      (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RbConfig::CONFIG["host_os"]) != nil
+    end
+
+    def OS.mac?
+      (/darwin/ =~ RbConfig::CONFIG["host_os"]) != nil
+    end
+
+    def OS.unix?
+      !OS.windows?
+    end
+
+    def OS.linux?
+      OS.unix? and not OS.mac?
+    end
+  end
+
+  class System
+    def initialize
+      @formatter = KPM::Formatter.new
+    end
+
+    def information(destination)
+      kpm_version = KPM::VERSION
+      kaui_version = get_kaui_version
+      killbill_version = get_killbill_version
+
+      environment = Hash[:kpm => {:system=>'KPM',:version => kpm_version},
+                         :kaui => {:system=>'Kaui',:version => kaui_version},
+                         :killbill => {:system=>'Killbill',:version => killbill_version}]
+
+      labels = [{:label => :system},
+                {:label => :version}]
+
+      @formatter.format(environment,labels)
+
+      show_environment_information
+      show_os_information
+      show_java_system_information
+      show_plugin_information(destination)
+    end
+
+    def show_environment_information
+      java_version = `java -version 2>&1`.split("\n")[0].split('"')[1]
```

Comment:
Might make sense to harden the code (e.g if java is not installed you will get an NPE, in which case we still want to output and for java we would 'no version found').